.evergreen/find-python3.sh

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/env bash
+#
+# find-python3.sh
+#
+# Usage:
+#   . /path/to/find-python3.sh
+#
+# This file defines the following utility functions:
+#   - is_python3
+#   - is_venv_capable
+#   - is_virtualenv_capable
+#   - find_python3
+# These functions may be invoked from any working directory.
+
+# is_python3
+#
+# Usage:
+#   is_python3 python
+#   is_python3 /path/to/python
+#
+# Parameters:
+#   "$1": The name or path of the python binary to test.
+#
+# Return 0 (true) if the given argument "$1" is a Python 3 binary.
+# Return a non-zero value (false) otherwise.
+#
+# Diagnostic messages may be printed to stderr (pipe 2). Redirect to /dev/null
+# to silence these messages.
+is_python3() (
+  set -o errexit
+  set -o nounset
+  set -o pipefail
+
+  # Binary to use, e.g. "python".
+  local -r bin="${1:?'is_python3 requires a name or path of a python binary to test'}"
+
+  # Binary must be executable.
+  command -V "$bin" &>/dev/null || return 1
+
+  # Reject binaries that do not support argument "-V".
+  "$bin" -V &>/dev/null || return 1
+
+  # Expect an output of the form: "Python x.y.z".
+  # Note: Python 2 binaries output to stderr rather than stdout.
+  local -r version_output="$("$bin" -V 2>&1)"
+
+  # "Python x.y.z" -> "x.y.z"
+  local -r version_str="$(perl -lne 'print $1 if m/.*([0-9]+\.[0-9]+\.[0-9]*)/' -- <(printf "%s" "$version_output"))"
+
+  # Evaluate 3.0.0 <= x.y.z.
+  sort -CV -- <(printf "%s\n%s\n" "3.0.0" "$version_str")
+)
+
+# is_venv_capable
+#
+# Usage:
+#   is_venv_capable python
+#   is_venv_capable /path/to/python
+#
+# Parameters:
+#   "$1": The name or path of the python binary to test.
+#
+# Return 0 (true) if the given argument "$1" can successfully evaluate the
+# command:
+#   "$1" -m venv venv
+# Return a non-zero value (false) otherwise.
+#
+# Diagnostic messages may be printed to stderr (pipe 2). Redirect to /dev/null
+# to silence these messages.
+is_venv_capable() (
+  set -o errexit
+  set -o nounset
+  set -o pipefail
+
+  local -r bin="${1:?'is_venv_capable requires a name or path of a python binary to test'}"
+
+  # Use a temporary directory to avoid polluting the caller's environment.
+  local -r tmp="$(mktemp -d)"
+  trap 'rm -rf "$tmp"' EXIT
+
+  # Evaluate the result of this function.
+  "$bin" -m venv "$tmp" 1>&2
```

Comment:
I recall that in some broken environments python -m venv/virtualenv can succeed but the activate script fails. But no need to change anything now. Let's wait until we actually encounter such a failure before we try to workaround it.